### PR TITLE
Model API Merge Into Main

### PR DESCRIPTION
**Pull request by hipsterusername**
Goal: Get Model API into main, then extend with new API endpoints

### CODE REVIEW

File: invokeai/app/api/routers/models.py [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+    operation_id="list_models",
+    responses={200: {"model": ModelsList }},
+)
+async def list_models() -> ModelsList:
+    """Gets a list of models"""
+    models_raw = ApiDependencies.invoker.services.model_manager.list_models()
+    models = parse_obj_as(ModelsList, { "models": models_raw })
+    return models
+
+        # @socketio.on("requestSystemConfig")
```
Comment (lstein) [inline invokeai/app/api/routers/models.py:57]: Why is everything below this line commented out? Is it slated to be replaced, or a placeholder?

**Review by lstein — APPROVED** [review]
LGTM

File: invokeai/app/api/routers/models.py (hunk 1 above)
Comment (hipsterusername) [reply invokeai/app/api/routers/models.py:57]: Yes - to be replaced. This is legacy SocketIO code to be referenced in finishing out the APIs.

**Comment by hipsterusername** [comment]
Once this gets merged in, I can attempt to add the model APIs again... 🙈

**Review by Kyle0654 — APPROVED** [review]

**Pull request merged**